### ISSUE TITLE
not valid DEPRECATION WARNING in template 

### ISSUE THREAD

**Issue by nicolas-g** [ansible/ansible#4514]
I keep getting this warnings for deprecated var formats that has nothing to do with Ansible vars but are part of my OS system template var : 
### configuration

```
ansible --version
ansible 1.4 (devel e131de4de0) last updated 2013/10/15 13:19:30 (GMT +300)

CentOS 5
Linux 2.6.34.7-56.40.amzn1.x86_64 #1 SMP Fri Oct 22 18:48:49 UTC 2010 x86_64 x86_64 x86_64 GNU/Linux

boto version: 2.13.3
```
### task

```
- name: environment setup
  template: src=etc-profiled-custom.sh.j2 dest=/etc/profile.d/custom.sh
```
### roles/comon/tempaltes/etc-profiled-custom.sh.j2

```
export CM_HOME=/var/cm
export CM_UTIL=/var/cm/bin

export JAVA_HOME=/usr/java/latest
export CATALINA_HOME=/opt/tomcat

export PLATFORM_BASE_DIR=/var/cm

export CM_MAVEN=/usr/local/maven/bin

export CM_ANT=/usr/local/ant/bin

export PATH="\$PATH:\$JAVA_HOME/bin:/usr/local/jmxclient:\$CM_UTIL:/usr/local/cm:\$CATALINA_HOME/bin:\$CM_MAVEN:\$CM_ANT"
```
### warning message

```
TASK: [common | environment setup] *****************************************
 [DEPRECATION WARNING]: Legacy variable subsitution, such as using ${foo} or
$foo instead of {{ foo }} is currently valid but will be phased out and has
been out of favor since version 1.2. This is the last of legacy features on our
deprecation list. You may continue to use this if you have specific needs for
now. This feature will be removed in version 1.6. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.

 [DEPRECATION WARNING]: Legacy variable subsitution, such as using ${foo} or
$foo instead of {{ foo }} is currently valid but will be phased out and has
been out of favor since version 1.2. This is the last of legacy features on our
deprecation list. You may continue to use this if you have specific needs for
now. This feature will be removed in version 1.6. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
```

**Comment by mpdehaan** [comment]
This happens when a template substitution can replace a $FOO variable because something is in scope.

I'll test and verify but I would think there's a chance there's an Ansible variable it is replacing.

Will look.

**Comment by mpdehaan** [comment]
Ok, I tracked this down.

The problem was because it is replacing occurances of \$ with $ because you have escaped them in the source file.

The workaround was to not escape them in the source file (there's no need unless legacy vars are enabled and you are afraid of replacements), however, I fixed the detector as well -- it now detects legacy change by a change in the number of dollar signs in the file, which is more accurate.
